src/main/java/nextstep/ladder/domain/Name.java

```diff
@@ -0,0 +1,38 @@
+package nextstep.ladder.domain;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.unmodifiableList;
+
+public class Name {
+
+    protected static final String ILLEGAL_LENGTH_ERROR_MESSAGE = "이름의 길이는 1이상 5이하만 가능합니다.";
+    protected static final int MIN_LENGTH = 1;
+    protected static final int MAX_LENGTH = 5;
+
+    private final String name;
+
+    public static List<Name> listOf(List<String> names) {
+        return unmodifiableList(
+                names.stream()
+                        .map(Name::new)
+                        .collect(Collectors.toList())
```

Comment:
다음처럼도 표현이 가능하는 점 단순 참고차 말씀드립니다.
```suggestion
                        .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
```

Reply:
오 이런 방법도 있었네요! 감사합니다